atlp-rwanda/e-commerce-serge-30-bn
# ft(RBAC): Implement role-based access control (RBAC) for user management[#187364865]

## ℹ️ Description

Implemented role assignment and permission management functionality to enable secure access control for various user actions. Defined user roles, including admin, user, and guest.

Fixes #(issue)

## 🧪 How can this be tested?

To test the changes implemented in this PR:

1. Create user accounts with different roles (admin, user, guest).
2. Verify that only users with the appropriate roles can access restricted functionalities.
3. Test assigning and removing roles for users via the role management interface.
4. Verify that role changes are reflected in user permissions and access control.

## 📍 Type of change

Please check the options that are relevant.


## ✅ Checklist:
